### PR TITLE
Fix #20027: Looping coaster with the reverse train option makes the wrong sound

### LINKED ISSUE #20027: Looping coaster whit the reverse train option make the wrong sound
### Operating System

Windows 11, 64-Bits

### OpenRCT2 build

OpenRCT2, v0.4.4-168-g6dfc622 (6dfc622 on develop) provided by GitHub

### Base game

RollerCoaster Tycoon 2

### Area(s) with this issue?

_No response_

### Describe the issue

It seem that the looping coaster dose the wrong sound when going reverse (dosen't happens foward or whit another ride/train whit same sounds as the looping normally do), it seem to replicate a way slower chainlift sound, when going backward.

### Steps to reproduce

1. Open any parks
2. Choose the looping coaster (the rct2 as it is the tested one as I report it)
3. build a bit of track going, and slowly gaining speed
4. reverse the train
5. let it go (bug will happen)

### Attachments

_No response_

### PR DESCRIPTION
**Pull request by kyphii**
Fix #20027

Previously, the lead car referenced Cars[vehicle_type] to get the friction sound. With the newly added reversed trains option, the lead car is not always type 0, so it can get the friction_sound_id of the other entries. This caused issues for the Looping Coaster, which does not initialize the friction_sound_id for the other entries, so it defaults to SoundId::LiftClassic (For the record, this bug could also be observed if you used the Ride Vehicle Editor to change the type of the lead car).

This is fixed by simply always referencing the front car type when retrieving the friction sound. This shouldn't impact anything else as far as I know (If it does, I'll come up with a more robust solution).

### CODE REVIEW

**Comment by Broxzier** [comment]
The 1000 ticks replay fails, but only on Windows builds, and Linux using clang. Is the sound that gets played saved in an entity? If so, this makes sense, but then I'm wondering why all other builds passed.

**Comment by kyphii** [comment]
I'm also seeing build issues for Windows and Linux on my other branches. Probably unrelated?

**Comment by duncanspumpkin** [comment]
> The 1000 ticks replay fails, but only on Windows builds, and Linux using clang. Is the sound that gets played saved in an entity? If so, this makes sense, but then I'm wondering why all other builds passed.

Yes the sound played is saved to the entity. So this is saying you are changing the output of other rides.

**Comment by kyphii** [comment]
Ok, will try a different approach then.

**Comment by karst** [comment]
Probably better to change the sound of the vehicles themselves, right? It's odd that not all entities on a train inherit the audio properties from car 0?

**Comment by kyphii** [comment]
> Probably better to change the sound of the vehicles themselves, right? It's odd that not all entities on a train inherit the audio properties from car 0?

There might be more train types than just the looping RC that are also set up improperly. If we changed them all, it would probably also trip the replay that's failing, I'd imagine (It'd be nice to know exactly what is failing in the replay now)

**Comment by duncanspumpkin** [comment]
> 

You can check the desync log if you run the replay locally. (needs to be windows x64 though as its tied to what computer arch recorded the replay for the desync log)

**Comment by Gymnasiast** [comment]
> Probably better to change the sound of the vehicles themselves, right?

You can generally bet on it that any custom vehicle is created by copying an existing one. In other words, we could fix the Looping RC trains, but all custom trains that are based on it will be broken.

> It's odd that not all entities on a train inherit the audio properties from car 0?

Custom content does all kinds of stuff that makes me cry into my beer, best not to go there ;-)

**Comment by spacek531** [comment]
> You can generally bet on it that any custom vehicle is created by copying an existing one. In other words, we could fix the Looping RC trains, but all custom trains that are based on it will be broken.

Rides made with buggy's ride maker need to be based on existing trains, but ones made with X7123M3-256 do not. I *generally* think I'm pretty good about filling in the sound on all vehicles but I'm sure someone will eventually find the one I forgot to.

**Comment by spacek531** [comment]
IMO patching it in RideObject.cpp is the better way to go about this, since this change has the potential to affect hacked rides.

File: src/openrct2/ride/Vehicle.cpp [hunk 1]
```diff
@@ -5410,7 +5410,9 @@ void Vehicle::UpdateSound()
     if (rideEntry == nullptr)
         return;
 
-    const auto& carEntry = rideEntry->Cars[vehicle_type];
+    // Always use the head car's sound data (Some of the other vehicle subtypes have improperly set data)
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp:5413]: We should probably add a note to the CarEntry struct that `friction_sound_id` and `CAR_ENTRY_FLAG_RIDERS_SCREAM` are only valid for the FrontCar/DefaultCar

**Comment by kyphii** [comment]
New implementation solves the replay issue. LMK if you folks still have a better idea though

**Comment by duncanspumpkin** [comment]
Will require a network version bump. Think thats about everything.

**Comment by X123M3-256** [comment]
> You can generally bet on it that any custom vehicle is created by copying an existing one.

Very few are. This would apply only to objects created with Buggy's ridemaker, but that was such an enormous pain to use and so limited that very few tracked rides were created that way, even when it was the only tool available. The most well known examples are the Amazing Earl rides, and those are probably the only ones that are still used. JAGIGLT used to be fairly popular but I haven't seen it in a long time, there are far better options available now.

Nothing that I or SpaceK made was based on an existing vehicle - my tools aren't even capable of doing that - though it's still entirely possible to make the same mistake.

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**